crates/stages/src/stages/index_account_history.rs

```diff
@@ -357,4 +362,90 @@ mod tests {
             ])
         );
     }
+
+    struct IndexAccountHistoryTestRunner {
+        pub(crate) tx: TestTransaction,
+        commit_threshold: u64,
+    }
+
+    impl Default for IndexAccountHistoryTestRunner {
+        fn default() -> Self {
+            Self { tx: TestTransaction::default(), commit_threshold: 1000 }
+        }
+    }
+
+    impl StageTestRunner for IndexAccountHistoryTestRunner {
+        type S = IndexAccountHistoryStage;
+
+        fn tx(&self) -> &TestTransaction {
+            &self.tx
+        }
+
+        fn stage(&self) -> Self::S {
+            Self::S { commit_threshold: self.commit_threshold }
+        }
+    }
+
+    impl ExecuteStageTestRunner for IndexAccountHistoryTestRunner {
+        type Seed = ();
+
+        fn seed_execution(&mut self, input: ExecInput) -> Result<Self::Seed, TestRunnerError> {
```

Comment:
I think what we want here is to seed the database with some random blocks and account changesets, as we do in e.g. Merkle stage: https://github.com/paradigmxyz/reth/blob/bb57556a6ab84f1c9f09eebd1f618220430b5101/crates/stages/src/stages/merkle.rs#L472

Then, in `validate_execution` we should dynamically calculate the shards from database data, and compare it with the contents of `AccountHistory` table.

Reply:
@shekhirin updated! PTAL